### DISCUSSION TITLE
Victron Smart Shunt recognized as Sinowelth Serialbattery

### DISCUSSION THREAD

**Discussion by claib** [Louisvdw/dbus-serialbattery#310] [Q&A] [answered]
hello there
after a reboot of the system, all of a sudden my vicreon smartshunt is regognized as Serialbattery ( Sinowelth ) BMS...
You can imagine the mess of false alerts...
My JK-BMS is recognized perfectly, also the Victron MPPT
What went wrong ?
Whant to do ?

Thx a lot

**Comment by Louisvdw** [comment, marked as answer]
Hi
I've seen that happen a few times.
If you remove the smartshunt communication wire and plug it back in it should get the correct driver and remember it from there.

It's related to issue https://github.com/Louisvdw/dbus-serialbattery/issues/150

**Reply by claib** [reply]
thx a lot !
tryed several time, but the Sinowealth keeps showing up and messing around.
changed USB ports, cables, reboot...
the smartshunt is still taken for a sinowealth serial battery.
any advice ?

**Reply by Louisvdw** [reply]
Are you using v0.13? Or one of the beta versions?
You can manually comment out the sinowealth line which will stop that driver being tested. For v0.13 that would be in dbus-serialbattery.py but in the latest beta the battery type list has moved to utils.py
Just add # in front to comment out the line (there are examples like mnb)

**Reply by SanderV4n** [reply]
@Louisvdw I agree the sinowealth driver should be disabled by default (for now). The main problem is that the protocol does not have any framing for the response message, its only 4 data bytes and one checksum/parity/something byte. I have tried to reverse engineer this additional checksum byte, which resulted in banging my head against the wall for some evenings and no progress.

For now it means that any random bytes received will be valid responses on the sinowealth commands. This could be from noise or from a device sending data. I also have a smartshunt which is now and then detected as a sinowealth bms, so also for me its a pain in the behind. When I find some more time Ill have an other go at it.

**Reply by claib** [reply]
shame on me ! It was the power supply.
Because my Raspi 3 was USB Powered with a QC Standard Power Supply, which was just enough, when i added the duppa.net 4 Way isolated TTL converter, it came to this strange behaviour.
![grafik](https://user-images.githubusercontent.com/8424229/205458934-488f375a-7c73-47e8-b067-a517f21c903b.png)
Changed the Power supply to a 4Amps Model, and now it works like a charm.
thx a lot !
